### ISSUE TITLE
.fit method could have an argument for deleting broken files

### ISSUE THREAD

**Issue by emailic** [keras-team/keras#16880]
**System information**.

TensorFlow version (you are using): 2.8.2
Are you willing to contribute it (Yes/No) : yes

**Describe the feature and the current behavior/state**.
Currently, when trying to fit a VGG16 model with a dataset containing some broken image files, the method simply returns an UnidentifiedImageError by PIL. Instead, it would be good to include an argument into the fit method which allows us to decide whether we would like to have the broken files deleted. 

<img width="452" alt="unidentif image erorr" src="https://user-images.githubusercontent.com/83206159/183615409-4cd34997-a188-4efd-ab70-c45a864cf6fb.png">

**Will this change the current api? How?**
Yes. .fit method will now have an argument delete_broken which is False by defualt, but when True, the broken image files are first deleted, and then the model is fitted. This is a way to handle UnidentifiedImageError which might confuse some users.

**Who will benefit from this feature?**
People who are training a model on custom data and may not be aware that some files are broken or dysfunctional. 

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing): I'd like to include an argument, let's say: delete_broken which takes on boolean values, and when True, deletes the broken image files, before executing the actual fitting of the dataset to the model.  When False, the .fit method behaviour stays the same as it is now: it raises PIL's UnideitifiedImageError

**Comment by tilakrayal** [comment]
@emailic,
Could you please share a reproducible code that supports your statement so that the issue can be easily understood? Thank you!

**Comment by emailic** [comment]
Hi @tilakrayal ,
thanks for getting back to me. 

`image_gen_train = ImageDataGenerator(rescale = ./255)`
`train_data_gen = image_gen_train.flow_from_directory(batch_size = batch_size,
directory = train_dir,
shuffle= True,
target_size = (IMG_SHAPE,IMG_SHAPE),
class_mode = 'categorical')`

`image_generator_validation = ImageDataGenerator(rescale=1./255)`
`val_data_gen= image_generator_validation.flow_from_directory(batch_size=batch_size,
directory=valid_dir,
target_size=(IMG_SHAPE, IMG_SHAPE),
class_mode='categorical')`

`vgg_classifier= model.fit(train_data_gen,
steps_per_epoch=(6610//batch_size),
epochs = 5,
validation_data=val_data_gen,
validation_steps=(1531//batch_size),
batch_size = batch_size,
workers=16,
use_multiprocessing=True,
verbose = 1)`

This is the code. In order to reproduce the error statement, you also need an Image dataset containing some broken (unreadable) files.

**Comment by chenmoneygithub** [comment]
Thanks for reporting the issue! You can actually use`tf.data.experimental.ignore_errors` to skip broken samples.

**Comment by emailic** [comment]
Thanks for getting back to me @chenmoneygithub ! Sounds like a good solution. Take care, Ema

**Comment by gowthamkpr** [comment]
@emailic Closing this issue as it has been resolved by the above [comment](https://github.com/keras-team/keras/issues/16880#issuecomment-1212257318). Thanks!
